components/binlog_utils_udf/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2023 Percona LLC and/or its affiliates. All rights reserved.
+
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 2 of
+# the License.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+option(WITH_BINLOG_UTILS_UDF "Build Binlog Utils component" ON)
+
+if(WITH_BINLOG_UTILS_UDF)
+  message(STATUS "Building Binlog Utils UDF component")
+else()
+  message(STATUS "Not building Binlog Utils UDF component")
+  return()
+endif()
+
+# We are not interesting in profiling tests.
+DISABLE_MISSING_PROFILE_WARNING()
+
+MYSQL_ADD_COMPONENT(binlog_utils_udf
+  binlog_utils_udf.cc
+  LINK_LIBRARIES extra::rapidjson
+  MODULE_ONLY
+)
+
+target_compile_definitions(component_binlog_utils_udf PRIVATE MYSQL_SERVER)
```

Comment:
Could you check if this `MYSQL_SERVER` define is indeed needed? After the binlog event stream refactoring you did something might have changed.

Reply:
It doesn't wort without `MYSQL_SERVER`:
```
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc: In function ‘{anonymous}::log_event_ptr {anonymous}::find_first_event(std::string_view)’:
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:113:3: error: ‘Binlog_file_reader’ was not declared in this scope; did you mean ‘Basic_binlog_file_reader’?
  113 |   Binlog_file_reader reader(false /* do not verify checksum */);
      |   ^~~~~~~~~~~~~~~~~~
      |   Basic_binlog_file_reader
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:114:7: error: ‘reader’ was not declared in this scope; did you mean ‘read’?
  114 |   if (reader.open(search_file_name, 0))
      |       ^~~~~~
      |       read
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:117:54: error: ‘reader’ was not declared in this scope; did you mean ‘read’?
  117 |   binlog::Decompressing_event_object_istream istream{reader};
      |                                                      ^~~~~~
      |                                                      read
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:117:60: error: no matching function for call to ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(<brace-enclosed initializer list>)’
  117 |   binlog::Decompressing_event_object_istream istream{reader};
      |                                                            ^
In file included from /data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:34:
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:115:3: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(const Transaction_payload_log_event&, Fde_ref_t)’
  115 |   Decompressing_event_object_istream(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:115:3: note:   candidate expects 2 arguments, 1 provided
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:99:3: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(const Tple_ptr_t&, Fde_ref_t)’
   99 |   Decompressing_event_object_istream(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:99:3: note:   candidate expects 2 arguments, 1 provided
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:85:12: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(IBasic_binlog_file_reader&)’
   85 |   explicit Decompressing_event_object_istream(
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:85:12: note:   conversion of argument 1 would be ill-formed:
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc: In function ‘{anonymous}::log_event_ptr {anonymous}::find_last_event(std::string_view)’:
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:135:3: error: ‘Binlog_file_reader’ was not declared in this scope; did you mean ‘Basic_binlog_file_reader’?
  135 |   Binlog_file_reader reader(false /* do not verify checksum */);
      |   ^~~~~~~~~~~~~~~~~~
      |   Basic_binlog_file_reader
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:136:7: error: ‘reader’ was not declared in this scope; did you mean ‘read’?
  136 |   if (reader.open(search_file_name, 0))
      |       ^~~~~~
      |       read
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:146:54: error: ‘reader’ was not declared in this scope; did you mean ‘read’?
  146 |   binlog::Decompressing_event_object_istream istream{reader};
      |                                                      ^~~~~~
      |                                                      read
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:146:60: error: no matching function for call to ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(<brace-enclosed initializer list>)’
  146 |   binlog::Decompressing_event_object_istream istream{reader};
      |                                                            ^
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:115:3: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(const Transaction_payload_log_event&, Fde_ref_t)’
  115 |   Decompressing_event_object_istream(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:115:3: note:   candidate expects 2 arguments, 1 provided
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:99:3: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(const Tple_ptr_t&, Fde_ref_t)’
   99 |   Decompressing_event_object_istream(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:99:3: note:   candidate expects 2 arguments, 1 provided
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:85:12: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(IBasic_binlog_file_reader&)’
   85 |   explicit Decompressing_event_object_istream(
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:85:12: note:   conversion of argument 1 would be ill-formed:
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc: In function ‘{anonymous}::log_event_ptr {anonymous}::find_previous_gtids_event(std::string_view)’:
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:170:3: error: ‘Binlog_file_reader’ was not declared in this scope; did you mean ‘Basic_binlog_file_reader’?
  170 |   Binlog_file_reader reader(false /* do not verify checksum */);
      |   ^~~~~~~~~~~~~~~~~~
      |   Basic_binlog_file_reader
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:171:7: error: ‘reader’ was not declared in this scope; did you mean ‘read’?
  171 |   if (reader.open(search_file_name, 0))
      |       ^~~~~~
      |       read
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:182:54: error: ‘reader’ was not declared in this scope; did you mean ‘read’?
  182 |   binlog::Decompressing_event_object_istream istream{reader};
      |                                                      ^~~~~~
      |                                                      read
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:182:60: error: no matching function for call to ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(<brace-enclosed initializer list>)’
  182 |   binlog::Decompressing_event_object_istream istream{reader};
      |                                                            ^
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:115:3: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(const Transaction_payload_log_event&, Fde_ref_t)’
  115 |   Decompressing_event_object_istream(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:115:3: note:   candidate expects 2 arguments, 1 provided
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:99:3: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(const Tple_ptr_t&, Fde_ref_t)’
   99 |   Decompressing_event_object_istream(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:99:3: note:   candidate expects 2 arguments, 1 provided
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:85:12: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(IBasic_binlog_file_reader&)’
   85 |   explicit Decompressing_event_object_istream(
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:85:12: note:   conversion of argument 1 would be ill-formed:
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc: In function ‘{anonymous}::log_event_ptr {anonymous}::find_last_gtid_event(std::string_view)’:
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:224:3: error: ‘Binlog_file_reader’ was not declared in this scope; did you mean ‘Basic_binlog_file_reader’?
  224 |   Binlog_file_reader reader(false /* do not verify checksum */);
      |   ^~~~~~~~~~~~~~~~~~
      |   Basic_binlog_file_reader
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:225:7: error: ‘reader’ was not declared in this scope; did you mean ‘read’?
  225 |   if (reader.open(search_file_name, 0))
      |       ^~~~~~
      |       read
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:236:54: error: ‘reader’ was not declared in this scope; did you mean ‘read’?
  236 |   binlog::Decompressing_event_object_istream istream{reader};
      |                                                      ^~~~~~
      |                                                      read
/data/mysql-server/8.x/components/binlog_utils_udf/binlog_utils_udf.cc:236:60: error: no matching function for call to ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(<brace-enclosed initializer list>)’
  236 |   binlog::Decompressing_event_object_istream istream{reader};
      |                                                            ^
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:115:3: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(const Transaction_payload_log_event&, Fde_ref_t)’
  115 |   Decompressing_event_object_istream(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:115:3: note:   candidate expects 2 arguments, 1 provided
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:99:3: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(const Tple_ptr_t&, Fde_ref_t)’
   99 |   Decompressing_event_object_istream(
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:99:3: note:   candidate expects 2 arguments, 1 provided
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:85:12: note: candidate: ‘binlog::Decompressing_event_object_istream::Decompressing_event_object_istream(IBasic_binlog_file_reader&)’
   85 |   explicit Decompressing_event_object_istream(
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/mysql-server/8.x/sql/binlog/decompressing_event_object_istream.h:85:12: note:   conversion of argument 1 would be ill-formed:
```